hercules-ci-agent/hercules-ci-agent/Hercules/Agent/Config.hs

```diff
@@ -68,17 +89,27 @@ readConfig loc = case loc of
 
 finalizeConfig :: Config 'Input -> IO (Config 'Final)
 finalizeConfig input = do
+
+  baseDir <- case baseDirectory input of
+    Just x -> pure x
+    Nothing -> throwIO $ FatalError $ "You need to specify " <> show keyBaseDirectory <> " in the configuration file."
```

Comment:
```suggestion
    Nothing -> throwIO $ FatalError $ "You need to specify " <> show keyBaseDirectory <> " in `--config agent.toml`"
```

Reply:
Not sure. "Breaks" when user supplies their own filename.

Reply:
could do `<filename>.toml` but I think that user will understand that better than the previous wording

Reply:
also kept `agent.toml` as we do the same in a few locations so far

Reply:
I appreciate the latter.
Maybe it's not too hard to give the real value. That's may actually be useful (even if it's not an editable source)

Reply:
See https://github.com/hercules-ci/hercules-ci-agent/pull/82/commits/e5089506f80cf41c9416c76bd71aa98fa41f5231